src/components/Tree/Node.tsx

```diff
@@ -137,7 +137,19 @@ export const Node = ({
                                     ))}
                             </span>
                             {icon && <span>{icon}</span>}
-                            <span data-test-id="node-link-name">{name}</span>
+                            <span className="tw-flex tw-items-center" data-test-id="node-link-name">
+                                {name}
+                                {badge && (
+                                    <div
+                                        className={merge([
+                                            "tw-flex tw-justify-center tw-items-center tw-w-8 tw-h-5 tw-ml-2 tw-bg-box-neutral tw-rounded-full",
+                                            selected && "tw-bg-transparent",
+                                        ])}
+                                    >
+                                        <span>{badge}</span>
```

Comment:
```suggestion
                                       {badge}
```

Do we need a span here? if not we could leave it out.